rialto_airflow/harvest/sul_pub.py

```diff
@@ -101,6 +102,14 @@ def publications(host, key, per_page=1000, limit=None):
                 more = False
                 break
 
+            # guard against the same record coming back twice
+            # see: https://github.com/sul-dlss/rialto-airflow/issues/684
+
+            sulpub_id = record["sulpubid"]
+            if sulpub_id in sulpub_ids:
+                continue
+            sulpub_ids.add(sulpub_id)
```

Comment:
is there any danger that this runs into memory problems with a very large set of ids?

Reply:
I used the `sulpub.jsonl` from the most recent crawl to extract all the `sulpubid`s. This created a set of 455,685 unique values (out of the 726,236 total with duplicates). The memory footprint for the set looks like it is 36 MB, so I think we'll be OK?

The downside to doing it this way is that if we were to restart the sulpub harvesting task it would lose the memory of having already stored some of the publications. I'm going to separately explore how we could add sulpub to the deduplication task. Doing it there would mean the harvest task could be restarted any number of times.